diffusion_with_autoscaler/autoscaler.py

```diff
@@ -231,49 +232,81 @@ async def send_batch(self, batch: List[Tuple[str, _BatchRequestModel]], server_u
             self._responses.update(result)
         finally:
             if server_url in self._server_status:
-                # TODO - if the server returns an error, track that so
-                #  we don't send more requests to it
-                self._server_status[server_url] = True
+                async with self._lock:
+                    if self.batching == "streamed":
+                        # decrement the number of currently processed requests.
+                        self._server_status[server_url] -= 1
```

Comment:
Assert that it's not negative for sanity?

Reply:
Done !